CONTRIBUTING.md

```diff
@@ -40,7 +40,7 @@ $ rubocop -V
   around it.
 * Make sure the test suite is passing and the code you wrote doesn't produce
   RuboCop offenses (usually this is as simple as running `bundle exec rake`).
-* [Squash related commits together][5].
+* [Squash related commits together][5] (or use [`git commit --amend` with `git push --force`][8]).
```

Comment:
I don't see this as an alternative, but as a particular case. I think it's better to link to the official git docs, which has information about both techniques.

Reply:
Forgot the link, this one :point_right: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

Reply:
> I don't see this as an alternative, but as a particular case.

For me this is an alternative: you can squash commits or even don't create multiple commits via `--amend` and so don't squash.

It's another approach, but for the same problem, I guess.

OK, I've updated the link, thanks.

Reply:
Ok, I saw the update now. I personally find the current recommendation too vague. Since the following bullet point is "Open a pull request that relates to *only* one subject", this bullet point could be interpreted as "Squash everything into one commit". My personal best practice here would be something along the lines of:

"""
Make sure the history of commits is organized into independent (even though related) units of changes , each of them as minimal as possible and keeping CI green. See [link] for more about how to rewrite your git history to achieve this goal.
"""

Reply:
> Ok, I saw the update now. I personally find the current recommendation too vague. Since the following bullet point is "Open a pull request that relates to only one subject", this bullet point could be interpreted as "Squash everything into one commit". My personal best practice here would be something along the lines of:

Sorry, I didn't notice details about "related commits" and that multiple-commits PRs are allowed. I'm personally used to single-commit PRs. And I don't like long descriptions about relevance and CI check for each commit in PR… So, I've just tuned my change to single-commit PRs case. Is it OK?

Reply:
I don't know, it all comes down to what the owners of the project consider the best thing to do for contributors. I'm only giving my personal practice, but I don't have a vote here :). Also, with the "squash" merge option, maybe this is actually not necessary :man_shrugging:.